integration/test-mb/test-amqp/src/test/java/org/wso2/ei/mb/test/amqp/AcknowledgementsTestCase.java

```diff
@@ -0,0 +1,308 @@
+                clientConfigPropertiesMap);
+        queueReceiver.registerSubscriber();
+
+        TimeUnit.SECONDS.sleep(50L);
+
+        // Creating a JMS publisher client configuration
+        QueueSender queueSender = new QueueSender("clientAckTestQueue", JMSAcknowledgeMode.CLIENT_ACKNOWLEDGE,
+                clientConfigPropertiesMap);
+        queueSender.sendMessages(SEND_COUNT, "text message");
+
+        TimeUnit.SECONDS.sleep(50L);
+
+        // Creating a second JMS consumer with client ack mode
+        QueueReceiver queueReceiver2 = new QueueReceiver("clientAckTestQueue", JMSAcknowledgeMode.CLIENT_ACKNOWLEDGE,
+                clientConfigPropertiesMap);
+        queueReceiver2.registerSubscriber();
+
+        TimeUnit.SECONDS.sleep(50L);
+
+        long totalMessagesReceived = queueReceiver.receivedMessageCount() + queueReceiver2
+                .receivedMessageCount();
+
+        queueReceiver.closeReceiver();
+
+        queueReceiver2.closeReceiver();
+
+        queueSender.closeSender();
+
+        Assert.assertEquals(totalMessagesReceived, EXPECTED_COUNT, "Expected message count not received.");
+    }
+
+    /**
+     * In this method we just test a sender and receiver with acknowledgements
+     * 1. Create consumer client with duplicate acknowledge mode
+     * 2. Publisher sends {@link #SEND_COUNT} messages.
+     * 3. Consumer will receive {@link #EXPECTED_COUNT} or more messages.
+     *
+     * @throws JMSException
+     * @throws NamingException
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    @Test
+    public void duplicatesOkAcknowledgementsTest()
+            throws JMSException, NamingException, IOException, InterruptedException {
+
+        // Creating a initial JMS consumer client configuration
+        QueueReceiver queueReceiver = new QueueReceiver("dupOkAckTestQueue", JMSAcknowledgeMode.DUPS_OK_ACKNOWLEDGE,
+                clientConfigPropertiesMap);
+        queueReceiver.registerSubscriber();
+
+        TimeUnit.SECONDS.sleep(10L);
+
+        // Creating a JMS publisher client configuration
+        QueueSender queueSender = new QueueSender("dupOkAckTestQueue", JMSAcknowledgeMode.DUPS_OK_ACKNOWLEDGE,
+                clientConfigPropertiesMap);
+        queueSender.sendMessages(SEND_COUNT / 10, "text message");
+
+        TimeUnit.SECONDS.sleep(10L);
+
+        long totalMessagesReceived = queueReceiver.receivedMessageCount();
+
+        queueReceiver.closeReceiver();
+
+        queueSender.closeSender();
+
+        // Evaluating
+        Assert.assertTrue(totalMessagesReceived >= EXPECTED_COUNT / 10, "The number of received messages " +
+                "(" + totalMessagesReceived + ") should be equal or more than the amount sent");
+    }
+
+    /**
+     * Clean up after test case.
+     */
+    @AfterClass
+    public void cleanup() throws IOException {
+
+        // Stop server instance and clean up.
+        serverManager.stopServer();
```

Comment:
An improvement would be to also delete any resources added to the server during this test before shutdown.